packages/graphql-yoga/__tests__/accept-header.spec.ts

```diff
@@ -0,0 +1,138 @@
+import { ExecutionResult } from 'graphql'
+import { createSchema, createYoga, Repeater, Plugin } from 'graphql-yoga'
+
+describe('accept header', () => {
+  it('instruct server to return an event-stream', async () => {
+    const yoga = createYoga({
+      schema: createSchema({
+        typeDefs: /* GraphQL */ `
+          type Query {
+            ping: String
+          }
+        `,
+      }),
+    })
+
+    // NOTE: this fails with content-type: application/json and body instead of query params
+    // is that intended?
+    const response = await yoga.fetch(`http://yoga/graphql?query=query{ping}`, {
+      headers: {
+        accept: 'text/event-stream',
+      },
+    })
+    expect(response.headers.get('content-type')).toEqual('text/event-stream')
+    const iterator = response.body![Symbol.asyncIterator]()
+    const { value } = await iterator.next()
+    const valueStr = Buffer.from(value).toString('utf-8')
+    expect(valueStr).toContain(
+      `data: ${JSON.stringify({ data: { ping: null } })}`,
+    )
+  })
+
+  it('instruct server to return a multipart result', async () => {
+    const yoga = createYoga({
+      schema: createSchema({
+        typeDefs: /* GraphQL */ `
+          type Query {
+            ping: String
+          }
+        `,
+        resolvers: {
+          Query: { ping: () => 'pong' },
+        },
+      }),
+    })
+
+    // NOTE: this fails with content-type: application/json and body instead of query params
+    // is that intended?
+    const response = await yoga.fetch(`http://yoga/graphql?query=query{ping}`, {
+      headers: {
+        accept: 'multipart/mixed',
+      },
+    })
+    expect(response.headers.get('content-type')).toEqual(
+      'multipart/mixed; boundary="-"',
+    )
+    const iterator = response.body![Symbol.asyncIterator]()
+    const { value } = await iterator.next()
+    const valueStr = Buffer.from(value).toString('utf-8')
+    // TODO: This test started failing after I replaced request(yoga) with yoga.fetch()
+    expect(valueStr).toContain(`Content-Type: application/json; charset=utf-8`)
+    expect(valueStr).toContain(`Content-Length: 24`)
+    expect(valueStr).toContain(`${JSON.stringify({ data: { ping: null } })}`)
```

Comment:
Any idea why this passes with super-test but not with yoga.fetch?

Reply:
Probably it is the difference between how we send and Node receives it. So it might be better to avoid using `response.body` as an async iterable and use `.getReader` then see the entire received result.

Reply:
You know what. We can use `await response.text()` to fetch entire response since it is singular already then we don't need to deal with the order, parsing and so on.